source/write/bulk-write.txt

```diff
@@ -340,6 +362,258 @@ subscribing to the observable and using the following methods:
+is replaced with a new document.
+
+If multiple documents match the query filter specified in
+a ``ClientNamespacedReplaceOneModel`` instance, the operation replaces the
+first result. You can specify a sort order in a `ClientReplaceOneOptions
+<{+core-api+}/com/mongodb/client/model/bulk/ClientReplaceOneOptions.html>`__
+instance to apply an order to matched documents before the driver
+performs the replace operation, as shown in the following code:
+
+.. code-block:: scala
+
+   val options = ClientReplaceOneOptions
+     .clientReplaceOneOptions()
+     .sort(ascending("_id"))
+
+Perform the Bulk Operation
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+After you define a ``ClientNamespacedWriteModel`` instance for each
+operation you want to perform, pass a list of these instances to the
+client ``bulkWrite()`` method. By default, the method runs the
+operations in the order they're specified.
+
+The following example performs multiple write operations by using the
+``bulkWrite()`` method:
+
+.. io-code-block::
+   :copyable: 
+
+   .. input:: /includes/write/client-bulk-write.scala
+      :start-after: start-perform
+      :end-before: end-perform
+      :language: scala
+      :dedent:
+
+   .. output::
+      
+      AcknowledgedSummaryClientBulkWriteResult{insertedCount=1, matchedCount=1, ...}
+
+If any of the write operations fail, the driver raises a
+``ClientBulkWriteException`` and does not perform any further individual operations.
+``ClientBulkWriteException`` includes a ``BulkWriteError`` that can be accessed by using the
+``ClientBulkWriteException.getWriteErrors()`` method, which provides details of the
+individual failure.
+
+.. _scala-client-bulk-write-options:
+
+Customize Bulk Write
+~~~~~~~~~~~~~~~~~~~~
+
+You can pass an instance of ``ClientBulkWriteOptions`` to the
+``bulkWrite()`` method to customize how the driver performs the bulk write
+operation.
+
+Order of Execution
+``````````````````
+
+By default, the driver runs the individual operations in a bulk
+operation in the order that you specify them until an error occurs, or
+until the operation completes successfully.
+
+However, you can pass ``false`` to the ``ordered()`` method when creating
+a ``ClientBulkWriteOptions`` instance to direct the driver to perform
+write operations in an unordered way. When using the unordered option,
+an error-producing operation does not prevent the driver from running
+other write operations in the bulk write operation.
+
+The following code sets the ``ordered`` option to ``false`` in an
+instance of ``ClientBulkWriteOptions`` and performs a bulk write operation to 
+insert multiple documents.
+
+.. literalinclude:: /includes/write/client-bulk-write.scala
+   :start-after: start-options
+   :end-before: end-options
+   :language: scala
+   :copyable:
+   :dedent:
+
+Even though the write operation inserting a document with a duplicate key results
+in an error, the other operations are performed because the write operation is
+unordered.
```

Comment:
```suggestion
Even though the write operation that inserts a document with a duplicate key results
in an error, the driver performs the other operations because the write operation is
unordered.
```